src/xrpld/app/misc/WasmHostFuncWrapper.cpp

```diff
@@ -95,11 +167,36 @@ getDataString(InstanceWrapper const* rt, int32_t src, int32_t ssz)
     return data;
 }
 
+#pragma push_macro("RET")
+#undef RET
 #define RET(x)                          \
     results->data[0] = WASM_I32_VAL(x); \
     results->num_elems = 1;             \
     return nullptr;
 
+// TODO: expand this to be more general, so that it can be used for all host
+// function wrappers
+#pragma push_macro("KEYLET_WRAPPER")
+#undef KEYLET_WRAPPER
+#define KEYLET_WRAPPER(FUNC, i, ...)                                      \
```

Comment:
Please use macro only for name concatenation, and  make the body as another  general function / template